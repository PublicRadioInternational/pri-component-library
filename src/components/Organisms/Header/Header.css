.siteHeader {
  background-color: white;
  border-bottom: 3px solid #ddd;
  display: flex;
  justify-content: space-between;
  padding: 0.5rem 0;
}

.logo {
  align-self: center;
  width: 113px;
}

.buttons {
  display: flex;
  justify-content: flex-end;
  width: 83%;
}

.button {
  margin-left: 0.75em;
}
