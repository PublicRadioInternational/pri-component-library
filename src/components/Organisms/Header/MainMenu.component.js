/**
 * @file MainMenu.component.js
 * Exports the main site menu.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './MainMenu.css';

/**
 * Component that renders the main site menu.
 */
export default class MainMenu extends Component {
  static propTypes = {
    toggleOpen: PropTypes.func
  };

  static defaultProps = {
    toggleOpen: false
  };

  constructor() {
    super();
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.toggleOpen(true);
  }

  render() {
    return (
      <div className={styles.mainMenu}>
        <div className={styles.wrapper}>
          <div className={styles.header}>
            <button className="close-drawer" onClick={this.handleClick}>
              <span className="oi oi-chevron-left" aria-label="close menu" />{' '}
              Close
            </button>
            <div className="text">
              <h3>
                <span className="pri icon-pri" aria-label="PRI" />
              </h3>
              <p>
                Our mission is to serve audiences as a distinctive content
                source for information, insights and cultural experiences
                essential to living in our diverse, interconnected world.
              </p>
              <div className="btn-group">
                <a
                  href="https://www.pri.org/about-pri"
                  className="btn btn-danger"
                >
                  <span
                    className="oi oi-info"
                    title="info"
                    aria-hidden="true"
                  />{' '}
                  Our Story
                </a>
                <a
                  href="https://www.pri.org/priorg-contact"
                  className="btn btn-danger"
                >
                  <span className="oi oi-envelope-closed" aria-hidden="true" />{' '}
                  Contact Us
                </a>
                <a
                  href="https://www.pri.org/donate/index.html?utm_source=navigation&amp;utm_medium=website&amp;utm_campaign=donations"
                  className="btn btn-danger"
                >
                  <span className="oi oi-heart" aria-hidden="true" /> Donate
                </a>
              </div>
            </div>
          </div>
          <div className="drawer-nav">
            <div
              id="accordion"
              className="accordion drawer-nav__accordion"
              role="tablist"
              aria-multiselectable="true"
            >
              <div className="card drawer-nav__card news-programs">
                <div
                  className="drawer-nav__card-header"
                  role="tab"
                  id="headingOne"
                >
                  <h5 className="mb-0">
                    <a
                      className="collapsed"
                      data-toggle="collapse"
                      data-parent="#accordion"
                      href="#collapseNews"
                      aria-expanded="true"
                      aria-controls="collapseNews"
                    >
                      News Programs{' '}
                      <span
                        className="oi oi-chevron-bottom"
                        aria-hidden="true"
                      />
                    </a>
                  </h5>
                </div>
                <div
                  id="collapseNews"
                  className="collapse"
                  role="tabpanel"
                  aria-labelledby="headingOne"
                >
                  <div className="card-block">
                    <ul className="menu">
                      <li className="item">
                        <a href="https://www.pri.org/programs/america-abroad">
                          America Abroad
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/it-happens">
                          As it Happens
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/boston-calling">
                          Boston Calling
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/day-6">Day 6</a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/innovation-hub">
                          Innovation Hub
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/living-earth">
                          Living on Earth
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/the-world">
                          PRI{"'"}s The World
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/takeaway">
                          The Takeaway
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div className="card drawer-nav__card music-variety-programs">
                <div
                  className="drawer-nav__card-header"
                  role="tab"
                  id="headingOne"
                >
                  <h5 className="mb-0">
                    <a
                      className="collapsed"
                      data-toggle="collapse"
                      data-parent="#accordion"
                      href="#collapseVariety"
                      aria-expanded="true"
                      aria-controls="collapseNews"
                    >
                      Music &amp; Variety Programs{' '}
                      <span
                        className="oi oi-chevron-bottom"
                        aria-hidden="true"
                      />
                    </a>
                  </h5>
                </div>
                <div
                  id="collapseVariety"
                  className="collapse"
                  role="tabpanel"
                  aria-labelledby="headingOne"
                >
                  <div className="card-block">
                    <ul className="menu">
                      <li className="item">
                        <a href="https://www.pri.org/programs/afropop-worldwide">
                          Afropop Worldwide
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/echoes">Echoes</a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/jazz-after-hours">
                          Jazz After Hours
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/live-wire">
                          Live Wire
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/on-story">
                          On Story
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/out-open">
                          Out in the Open
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/pittsburgh-symphony-orchestra">
                          Pittsburgh Symphony Orchestra
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/q-cbc">
                          q from the CBC
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/selected-shorts">
                          Selected Shorts
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/studio-360">
                          Studio 360
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div className="card drawer-nav__card podcasts">
                <div
                  className="drawer-nav__card-header"
                  role="tab"
                  id="headingOne"
                >
                  <h5 className="mb-0">
                    <a
                      className="collapsed"
                      data-toggle="collapse"
                      data-parent="#accordion"
                      href="#collapsePodcasts"
                      aria-expanded="true"
                      aria-controls="collapsePodcasts"
                    >
                      Podcasts{' '}
                      <span
                        className="oi oi-chevron-bottom"
                        aria-hidden="true"
                      />
                    </a>
                  </h5>
                </div>
                <div
                  id="collapsePodcasts"
                  className="collapse"
                  role="tabpanel"
                  aria-labelledby="headingOne"
                >
                  <div className="card-block">
                    <ul className="menu">
                      <li className="item">
                        <a href="https://www.pri.org/programs/lbjs-war">
                          LBJ{"'"}s War
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/otherhood">
                          Otherhood
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/podcast-playlist">
                          Podcast Playlist
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/science-happiness">
                          Science of Happiness
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/world-words">
                          The World in Words
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/things-go-boom">
                          Things That Go Boom
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div className="card drawer-nav__card listen">
                <div
                  className="drawer-nav__card-header"
                  role="tab"
                  id="headingOne"
                >
                  <h5 className="mb-0">
                    <a
                      className="collapsed"
                      data-toggle="collapse"
                      data-parent="#accordion"
                      href="#collapseListen"
                      aria-expanded="true"
                      aria-controls="collapseListen"
                    >
                      Listen{' '}
                      <span
                        className="oi oi-chevron-bottom"
                        aria-hidden="true"
                      />
                    </a>
                  </h5>
                </div>
                <div
                  id="collapseListen"
                  className="collapse"
                  role="tabpanel"
                  aria-labelledby="headingOne"
                >
                  <div className="card-block">
                    <ul className="menu">
                      <li className="item">
                        <a href="https://www.pri.org/listen">Live Stream</a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/podcasts-program">
                          Podcasts by Program
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/ways-listen">
                          Ways to Listen
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div className="card drawer-nav__card features">
                <div
                  className="drawer-nav__card-header"
                  role="tab"
                  id="headingTwo"
                >
                  <h5 className="mb-0">
                    <a
                      className="collapsed"
                      data-toggle="collapse"
                      data-parent="#accordion"
                      href="#collapseFeatures"
                      aria-expanded="false"
                      aria-controls="collapseFeatures"
                    >
                      Features{' '}
                      <span
                        className="oi oi-chevron-bottom"
                        aria-hidden="true"
                      />
                    </a>
                  </h5>
                </div>
                <div
                  id="collapseFeatures"
                  className="collapse"
                  role="tabpanel"
                  aria-labelledby="headingTwo"
                >
                  <div className="card-block">
                    <ul className="menu">
                      <li className="item">
                        <a href="https://www.pri.org/verticals/across-womens-lives">
                          Across Women{"'"}s Lives
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/verticals/global-nation">
                          Global Nation
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/verticals/global-security">
                          Global Security
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/verticals/livable-planet">
                          Livable Planet
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/programs/globalpost">
                          GlobalPost
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/collections/global-hit">
                          Global Hit
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/collections/global-satire">
                          Global Satire
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/music-heard-on-air">
                          Music Heard on Air
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div className="card drawer-nav__card categories">
                <div
                  className="drawer-nav__card-header"
                  role="tab"
                  id="headingTwo"
                >
                  <h5 className="mb-0">
                    <a
                      className="collapsed"
                      data-toggle="collapse"
                      data-parent="#accordion"
                      href="#collapseCategories"
                      aria-expanded="false"
                      aria-controls="collapseCategories"
                    >
                      Categories{' '}
                      <span
                        className="oi oi-chevron-bottom"
                        aria-hidden="true"
                      />
                    </a>
                  </h5>
                </div>
                <div
                  id="collapseCategories"
                  className="collapse"
                  role="tabpanel"
                  aria-labelledby="headingTwo"
                >
                  <div className="card-block">
                    <ul className="menu">
                      <li className="item">
                        <a href="https://www.pri.org/sections/arts-culture-media">
                          Arts &amp; Entertainment
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/conflict-justice">
                          Conflict &amp; Justice
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/culture">
                          Culture &amp; Society
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/development-education">
                          Development &amp; Education
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/health-medicine">
                          Health &amp; Medicine
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/history">
                          History
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/verticals/global-nation">
                          Immigration
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/business-economics-and-jobs">
                          Money
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/politics">
                          Politics
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/science-tech-environment">
                          Science &amp; Environment
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/sections/technology">
                          Technology
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/verticals/across-womens-lives">
                          Women &amp; Girls
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div className="card drawer-nav__card about">
                <div
                  className="drawer-nav__card-header"
                  role="tab"
                  id="headingThree"
                >
                  <h5 className="mb-0">
                    <a
                      className="collapsed"
                      data-toggle="collapse"
                      data-parent="#accordion"
                      href="#collapseAbout"
                      aria-expanded="false"
                      aria-controls="collapseAbout"
                    >
                      About PRI{' '}
                      <span
                        className="oi oi-chevron-bottom"
                        aria-hidden="true"
                      />
                    </a>
                  </h5>
                </div>
                <div
                  id="collapseAbout"
                  className="collapse"
                  role="tabpanel"
                  aria-labelledby="headingThree"
                >
                  <div className="card-block">
                    <ul className="menu">
                      <li className="item">
                        <a href="https://www.pri.org/about-pri">About Us</a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/financials">
                          Annual Report
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/priorg-contact">
                          Contact Us
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/financials">Financials</a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/team">
                          Meet the PRI.org team
                        </a>
                      </li>
                      <li className="item">
                        <a href="https://www.pri.org/work-us">Work with Us</a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div className="card drawer-nav__card giving">
                <div
                  className="drawer-nav__card-header"
                  role="tab"
                  id="headingThree"
                >
                  <h5 className="mb-0">
                    <a
                      className="collapsed"
                      data-toggle="collapse"
                      data-parent="#accordion"
                      href="#collapseGiving"
                      aria-expanded="false"
                      aria-controls="collapseGiving"
                    >
                      Giving{' '}
                      <span
                        className="oi oi-chevron-bottom"
                        aria-hidden="true"
                      />
                    </a>
                  </h5>
                </div>
                <div
                  id="collapseGiving"
                  className="collapse"
                  role="tabpanel"
                  aria-labelledby="headingThree"
                >
                  <div className="card-block">
                    <ul className="menu">
                      <li className="item">
                        <a href="https://www.pri.org/donate/index.html?utm_source=navigation&amp;utm_medium=website&amp;utm_campaign=donations">
                          Donate
                        </a>
                      </li>
                      <li className="item">
                        <a href="/">Donor Benefits</a>
                      </li>
                      <li className="item">
                        <a href="/">Annual Report</a>
                      </li>
                      <li className="item">
                        <a href="/">Donation FAQs</a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="drawer-footer">
            <ul className="social-menu">
              <li className="social-item facebook">
                <a
                  href="https://www.facebook.com/publicradiointernational"
                  title="Follow PRI on Facebook"
                >
                  <i className="ss-icon ss-social">Facebook</i>
                </a>
              </li>
              <li className="social-item twitter">
                <a href="https://twitter.com/pri" title="Follow PRI on Twitter">
                  <i className="ss-icon ss-social">Twitter</i>
                </a>
              </li>
              <li className="social-item newsletter">
                <a
                  href="https://www.pri.org/newsletters"
                  title="Subscribe to our Newsletters"
                >
                  <i className="oi oi-envelope-open" aria-label="Newsletters" />
                </a>
              </li>
              <li className="social-item rss">
                <a
                  href="https://www.pri.org/stories/feed/everything"
                  title="PRI RSS Feed"
                >
                  <i className="ss-icon ss-social">RSS</i>
                </a>
              </li>
              <li className="social-item login">
                <a href="https://www.pri.org/user/me" title="My Account">
                  <i className="oi oi-person" aria-label="Login" />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div className="antiSelect" />
        <div className="label" />
      </div>
    );
  }
}
