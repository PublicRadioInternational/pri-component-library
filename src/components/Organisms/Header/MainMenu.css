/* import colors... */
@value colors: "../../00_global/colors.css";
@value whiteDark, gray, grayLighter, grayDark, orange, orangeDark, blue, yellow, green from colors;

/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value heading from fonts;

/* import breakpoints */
@value min as bp-min from "../../00_global/breakpoints.css";

.wrapper {
  background-color: gray;
  flex-flow: column;
  min-height: 100vh;
  position: relative;
}

@media bp-min {
  .wrapper {
    display: flex;
    overflow-y: auto;
  }
}

.header {
  border-top: 3px solid orange;
  flex: 0 0 auto;
  padding: 15px 15px 20px;
  color: grayLighter;
}

.closeMenu {
  align-items: center;
  background: none;
  border: none;
  color: grayLighter;
  cursor: pointer;
  display: flex;
  float: right;
  font-size: 0.95rem;
  padding: 0;
}

.inlineLink {
  height: 20px;
  margin: 0;
  width: 20px;
}

.headerText {
  clear: both;
  padding-top: 20px;
}

.headerLogo {
  width: 36%;
}

.maxWidth {
  max-width: 100%;
}

/* Buttons */
.btnGroup {
  margin: 14px 0 2px;
}

@media bp-min {
  display: flex;
}

.btnHeaderLeft,
.btnHeaderMiddle,
.btnHeaderRight {
  display: block;
  font: normal 0.875rem/1.5 heading;
  letter-spacing: 0;
  margin-bottom: 0.3rem;
  padding: 0.275rem 0.5rem;
  text-transform: none;
}

@media bp-min {
  .btnHeaderLeft,
  .btnHeaderMiddle,
  .btnHeaderRight {
    display: inline-block;
    padding: 0.475rem 0.5rem;
  }

  .btnHeaderLeft {
    border-bottom-right-radius: 0;
    border-top-right-radius: 0;
  }

  .btnHeaderMiddle {
    border-left: 1px solid orangeDark;
    border-radius: 0;
  }

  .btnHeaderRight {
    border-bottom-left-radius: 0;
    border-top-left-radius: 0;
    border-left: 1px solid orangeDark;
  }
}

/* Accordion Nav */
.accordionMenu {
  background: whiteDark;
  flex: 1 1 auto;
  padding: 10px;
}

/* Social Links */
.socialMenu {
  background-color: gray;
  display: flex;
  flex: 0 1 57px;
  justify-content: space-between;
}

.socialMenuItem {
  display: inline-block;
  overflow: hidden;
}

.socialMenuLink {
  color: gray;
  display: block;
  padding: 1rem;
}

.socialMenuLinkFB:hover {
  background-color: #5179bb;
  color: #5179bb;
}

.socialMenuLinkTW:hover {
  background-color: #1da1f2;
  color: #1da1f2;
}

.socialMenuLinkEM:hover {
  background-color: orange;
}

.socialMenuLinkRSS:hover {
  background-color: green;
  color: green;
}

.socialMenuLinkLogin:hover {
  background-color: blue;
}

.socialMenuIcon {
  background-color: #ddd;
  border-radius: 50px;
  height: 16px;
  padding: 5px 3px 5px 1px;
  width: 21px;
}

.socialMenuIconFull {
  background: none;
  border-radius: 0;
  color: #ddd;
  height: 25px;
  padding: 0;
  width: 25px;
}

.socialMenuIconRSS {
  height: 23px;
  padding: 2px 2px 2px 3px;
  width: 22px;
}

.socialMenuIconFullLg {
  height: 37px;
  width: 33px;
}
