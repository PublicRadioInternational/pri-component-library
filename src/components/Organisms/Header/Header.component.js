/**
 * @file Header.component.js
 * Exports the site header.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Header.css';
import Icon from '../../Atoms/Svg/Icons.component';
import PriLogo from '../../Atoms/Svg/PriLogo.component';
import Dropdown from '../../Molecules/Dropdown/Dropdown.component';
import DropdownItem from '../../Atoms/DropdownItem/DropdownItem.component';
import Search from '../../Molecules/Search/Search.component';
import MainMenu from './MainMenu.component';

const cx = classNames.bind(styles);

/**
 * Component that renders the header.
 */
export default class Header extends Component {
  state = { isMenuOpen: false };

  toggleOpen = () => {
    this.setState(prevState => ({
      isMenuOpen: !prevState.isMenuOpen
    }));
  };

  render() {
    const { baseUrl } = this.props;
    const bgClasses = cx({
      bg: true,
      bgOpen: this.state.isMenuOpen
    });

    return (
      <header className={styles.siteHeader}>
        <button
          className={bgClasses}
          onClick={this.toggleOpen}
          aria-label="Close Overlay"
        />
        <div
          className={`${styles.mainMenu} ${this.state.isMenuOpen &&
            styles.mainMenuOpen}`}
        >
          <MainMenu toggleOpen={this.toggleOpen} />
        </div>
        <div className={styles.branding}>
          <button
            className={styles.menuBtn}
            onClick={this.toggleOpen}
            aria-label="Open Menu"
          >
            <Icon name="menu" className={styles.menuIcon} />
          </button>
          <a href="/" className={styles.logo}>
            <PriLogo title="PRI" className={styles.siteLogo} />
          </a>
        </div>

        <div className={styles.buttons}>
          <Dropdown
            url={`${baseUrl}/listen`}
            title="Listen"
            className={styles.button}
            iconClass={styles.mobileSmallIcon}
            small
            icon="play"
          >
            <DropdownItem url={`${baseUrl}/listen`}>Live Stream</DropdownItem>
            <DropdownItem url={`${baseUrl}/live/schedule`}>
              Live Stream Schedule
            </DropdownItem>
            <DropdownItem url={`${baseUrl}/podcasts-program`}>
              Podcasts by Program
            </DropdownItem>
            <DropdownItem url={`${baseUrl}/ways-listen`}>
              Ways to Listen
            </DropdownItem>
            <DropdownItem isHr />
            <DropdownItem url={`${baseUrl}/shows`}>
              All Programs & Podcasts
            </DropdownItem>
          </Dropdown>

          <Dropdown
            url={`${baseUrl}/newsletters`}
            title="Newsletters"
            className={`${styles.button} ${styles.newsletterBtn}`}
            iconClass={styles.mobileSmallIcon}
            small
            icon="envelope"
          >
            <DropdownItem heading="Daily Newsletters" />
            <DropdownItem url={`${baseUrl}/get-scan-your-inbox-each-weekday`}>
              The Scan
            </DropdownItem>
            <DropdownItem isHr />
            <DropdownItem heading="Weekly Newsletters" />
            <DropdownItem url={`${baseUrl}/subscribe-global-nation-newsletter`}>
              Global Nation
            </DropdownItem>
            <DropdownItem url={`${baseUrl}/subscribe-safemode-weekly-digest`}>
              Global Security
            </DropdownItem>
            <DropdownItem isHr />
            <DropdownItem url={`${baseUrl}/user/me/newsletters`}>
              Manage My Newsletters
            </DropdownItem>
          </Dropdown>

          <Dropdown
            url={`${baseUrl}/search/node`}
            title="Search"
            className={styles.button}
            iconClass={styles.mobileSmallIcon}
            small
            icon="search"
          >
            <DropdownItem>
              <Search baseUrl={baseUrl} />
            </DropdownItem>
          </Dropdown>

          <Dropdown
            url={`${baseUrl}/donate/about?utm_source=navigation&utm_medium=website&utm_campaign=donations`}
            title="Donate"
            color="Orange"
            className={styles.button}
            iconClass={styles.mobileSmallIcon}
            small
            icon="heart"
          >
            <DropdownItem
              url={`${baseUrl}/donate/general?utm_source=navigation&utm_medium=website&utm_campaign=donations`}
            >
              Give Now
            </DropdownItem>
          </Dropdown>
        </div>
      </header>
    );
  }
}

Header.propTypes = {
  baseUrl: PropTypes.string
};

Header.defaultProps = {
  baseUrl: 'https://www.pri.org'
};
