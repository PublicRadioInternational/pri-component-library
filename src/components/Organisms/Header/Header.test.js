/**
 * @file Header.test.js
 * Contains tests for Header.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header.component';

describe('<Header />', () => {
  it('Matches the Header snapshot', () => {
    const component = renderer.create(<Header />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
