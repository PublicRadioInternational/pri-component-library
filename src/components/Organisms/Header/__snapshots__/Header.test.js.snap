// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Matches the Header snapshot 1`] = `
<header
  className="siteHeader"
>
  <a
    className="logo"
    href="/"
  >
    <svg
      aria-hidden={null}
      className={null}
      fill="currentcolor"
      viewBox={undefined}
    >
      <title>
        PRI
      </title>
      <use
        xlinkHref="#undefined"
      />
    </svg>
  </a>
  <div
    className="buttons"
  >
    <div
      className="dropdownGrp button"
    >
      <a
        className="btnMobileWhite btnGrp"
        href="https://www.pri.org/listen"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="inlineSvg null"
          fill="currentcolor"
          viewBox={undefined}
        >
          <use
            xlinkHref="#undefined"
          />
        </svg>
        <span
          className="textLabel textLabelMobile"
        >
          Listen
        </span>
      </a>
      <button
        aria-expanded={null}
        aria-haspopup={true}
        aria-label="open menu"
        className="btnMobileWhite btnDropdownWhite btnHide"
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="dropdownGrp button"
    >
      <a
        className="btnMobileWhite btnGrp"
        href="https://www.pri.org/newsletters"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="inlineSvg null"
          fill="currentcolor"
          viewBox={undefined}
        >
          <use
            xlinkHref="#undefined"
          />
        </svg>
        <span
          className="textLabel textLabelMobile"
        >
          Newsletters
        </span>
      </a>
      <button
        aria-expanded={null}
        aria-haspopup={true}
        aria-label="open menu"
        className="btnMobileWhite btnDropdownWhite btnHide"
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="dropdownGrp button"
    >
      <a
        className="btnMobileWhite btnGrp"
        href="https://interactive-dev.pri.org/staging/prototypes/homepage/iteration-2.html#"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="inlineSvg null"
          fill="currentcolor"
          viewBox={undefined}
        >
          <use
            xlinkHref="#undefined"
          />
        </svg>
        <span
          className="textLabel textLabelMobile"
        >
          Search
        </span>
      </a>
      <button
        aria-expanded={null}
        aria-haspopup={true}
        aria-label="open menu"
        className="btnMobileWhite btnDropdownWhite btnHide"
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="dropdownGrp button"
    >
      <a
        className="btnMobileOrange btnGrp"
        href="https://interactive-dev.pri.org/staging/prototypes/homepage/iteration-2.html#"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="inlineSvg null"
          fill="currentcolor"
          viewBox={undefined}
        >
          <use
            xlinkHref="#undefined"
          />
        </svg>
        <span
          className="textLabel textLabelMobile"
        >
          Donate
        </span>
      </a>
      <button
        aria-expanded={null}
        aria-haspopup={true}
        aria-label="open menu"
        className="btnMobileOrange btnDropdownOrange btnHide"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</header>
`;
